src/include/sof/samples/audio/detect_test.h

```diff
@@ -39,6 +40,27 @@ struct sof_detect_test_config {
 	uint32_t reserved[1];
 } __attribute__((packed));
 
+uint16_t test_keyword_get_sample_valid_bytes(struct comp_dev *dev);
+
+uint32_t test_keyword_get_load_mips(struct comp_dev *dev);
+
+uint32_t test_keyword_get_detected(struct comp_dev *dev);
+int test_keyword_set_detected(struct comp_dev *dev, uint32_t detected);
+
+const int16_t *test_keyword_get_input(struct comp_dev *dev);
+
+int16_t test_keyword_get_input_byte(struct comp_dev *dev, uint32_t index);
+int16_t test_keyword_get_input_elem(struct comp_dev *dev, uint32_t index);
+int test_keyword_set_input_elem(struct comp_dev *dev, uint32_t index, int16_t val);
+
+size_t test_keyword_get_input_size(struct comp_dev *dev);
+int test_keyword_set_input_size(struct comp_dev *dev, size_t input_size);
+
+uint32_t test_keyword_get_drain_req(struct comp_dev *dev);
+int test_keyword_set_drain_req(struct comp_dev *dev, uint32_t drain_req);
```

Comment:
For other reviewers: We need to figure out if these getters and setters are okay.

Reply:
Is this purely a test API ?

Reply:
@lgirdwood no test API. We are using the existing `detect_test` component to plugin our own detection algorithm. For this we need to have some getters/setters in place.